pkg/grizzly/providers.go

```diff
@@ -6,40 +6,102 @@ import (
 
 	"github.com/gobwas/glob"
 	"github.com/grafana/tanka/pkg/kubernetes/manifest"
+	"gopkg.in/yaml.v3"
 )
 
 // Resource represents a single Resource destined for a single endpoint
-type Resource struct {
-	UID      string      `json:"uid"`
-	Filename string      `json:"filename"`
-	Handler  Handler     `json:"handler"`
-	Detail   interface{} `json:"detail"`
-	JSONPath string      `json:"path"`
+type Resource map[string]interface{}
+
+// NewResource returns a new Resource object
+func NewResource(apiVersion, kind, name string, spec map[string]interface{}) Resource {
+	resource := Resource{
+		"apiVersion": apiVersion,
+		"kind":       kind,
+		"metadata": map[string]interface{}{
+			"name": name,
+		},
+		"spec": spec,
+	}
+	return resource
 }
 
 // APIVersion returns the group and version of the provider of the resource
 func (r *Resource) APIVersion() string {
-	return r.Handler.APIVersion()
+	return (*r)["apiVersion"].(string)
 }
 
 // Kind returns the 'kind' of the resource, i.e. the type of the handler
 func (r *Resource) Kind() string {
-	return r.Handler.Kind()
+	return (*r)["kind"].(string)
+}
+
+func (r *Resource) Name() string {
+	return r.GetMetadata("name")
 }
 
 // Key returns a key that combines kind and uid
 func (r *Resource) Key() string {
-	return fmt.Sprintf("%s/%s", r.Kind(), r.UID)
+	return fmt.Sprintf("%s/%s", r.Kind(), r.Name())
 }
 
-// GetRepresentation Gets the string representation for this resource
-func (r *Resource) GetRepresentation() (string, error) {
-	return r.Handler.GetRepresentation(r.UID, *r)
+func (r *Resource) GetMetadata(key string) string {
+	metadata := (*r)["metadata"].(map[string]interface{})
+	return metadata[key].(string)
+}
+
+func (r *Resource) SetMetadata(key, value string) Resource {
+	metadata := (*r)["metadata"].(map[string]interface{})
+	metadata[key] = value
+	(*r)["metadata"] = metadata
+	return *r
+}
+
+func (r *Resource) GetSpecString(key string) string {
+	spec := (*r)["spec"].(map[string]interface{})
+	return spec[key].(string)
+}
+
+func (r *Resource) SetSpecString(key, value string) *Resource {
+	spec := (*r)["spec"].(map[string]interface{})
+	spec[key] = value
+	(*r)["spec"] = spec
+	return r
+}
+
+func (r *Resource) DeleteSpecKey(key string) *Resource {
+	spec := (*r)["spec"].(map[string]interface{})
+	delete(spec, key)
+	(*r)["spec"] = spec
+	return r
+}
+
+func (r *Resource) Spec() map[string]interface{} {
+	return (*r)["spec"].(map[string]interface{})
 }
 
-// GetRemoteRepresentation Gets the string representation for this resource
-func (r *Resource) GetRemoteRepresentation() (string, error) {
-	return r.Handler.GetRemoteRepresentation(r.UID)
+func (r *Resource) AsResourceList() ResourceList {
+	key := r.Key()
+	resources := ResourceList{}
+	resources[key] = *r
+	return resources
+}
+
+func (r *Resource) SpecAsJSON() (string, error) {
```

Comment:
No, because this just returns the spec element, as JSON, not the whole thing. `toJSON` could be fine, but what about `toYAML`, does that return the spec, or the whole resource?

Reply:
If Resource.Spec is a type, you could setup `func (s Spec) toJSON() (string,error)`